pkg/servicetest/servicetest_test.go

```diff
@@ -0,0 +1,135 @@
+package servicetest_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"path/filepath"
+
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/drud/ddev/pkg/dockerutil"
+	"github.com/drud/ddev/pkg/fileutil"
+	"github.com/drud/ddev/pkg/plugins/platform"
+	"github.com/drud/ddev/pkg/testcommon"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	TestSites = []testcommon.TestSite{
+		{
+			Name:                          "TestMainPkgDrupal8",
+			SourceURL:                     "https://github.com/drud/drupal8/archive/v0.6.0.tar.gz",
+			ArchiveInternalExtractionPath: "drupal8-0.6.0/",
+		},
+	}
+	ServiceFiles []string
+	ServiceDir   string
+)
+
+func TestMain(m *testing.M) {
+	for i := range TestSites {
+		err := TestSites[i].Prepare()
+		if err != nil {
+			log.Fatalf("Prepare() failed on TestSite.Prepare(), err=%v", err)
+		}
+	}
+
+	log.Debugln("Running tests.")
+	testRun := m.Run()
+
+	for i := range TestSites {
+		TestSites[i].Cleanup()
+	}
+
+	os.Exit(testRun)
+}
+
+func TestServicesSetup(t *testing.T) {
+	if !testing.Short() {
+		var err error
+		assert := assert.New(t)
+
+		ServiceDir, err = filepath.Abs("../../services")
+		assert.NoError(err)
+
+		err = filepath.Walk(ServiceDir, func(path string, f os.FileInfo, _ error) error {
+			if !f.IsDir() && strings.HasPrefix(f.Name(), "docker-compose") {
+				ServiceFiles = append(ServiceFiles, f.Name())
+			}
+			return nil
+		})
+		assert.NoError(err)
+	} else {
+		fmt.Println("services tests skipped in short mode")
+	}
+}
+
+// TestServices tests each service compose file in the services folder.
+// It tests that a site can fully start w/ the compose file present, and
+// checks that any exposed HTTP ports return 200.
+func TestServices(t *testing.T) {
+	assert := assert.New(t)
+
+	for _, service := range ServiceFiles {
+		fmt.Println("Service tests for ", service)
+
+		serviceName := strings.TrimPrefix(service, "docker-compose.")
+		serviceName = strings.TrimSuffix(serviceName, ".yaml")
+
+		for _, site := range TestSites {
```

Comment:
This could potentially cause a pretty large increase in test time as we add additional test sites and/or services. It feels like we could just do 1 site start here for each test site with ALL services enabled, and then test that each service is working? It feels like that would save us a significant amount of time on test runs as we add services.